### ISSUE TITLE
Option to self-host

### ISSUE THREAD

**Issue by appetrosyan** [musescore/MuseScore#23610]
### Your idea

I'd like to have the option to use MuseScore without any of its online "features". That, is, specifically a build which either offers the option to self-host most of the features that would require access to the internet, or the option to remove them from the UI. 

### Problem to be solved

Most people that prefer Free and Open Source software, also prefer the ethos of those programs. While they optionally provide online features, those are rarely to be considered part of the core package, and rarer still require registration and data handling. 

The fact that the Musescore binary is an open source front-end to services that may become proprietary is quite concerning. It also comes with the artificially imposed limitation of not being able to know which features require the internet, (which is not always a given, not always available, and oftentimes unreliably slow), I'd like to just have access to just the binary that only processes data locally. 

This has an added security benefit, because even allegedly open source software when hosted somewhere else, and centralised offers none of the security benefits of Free and Open Source software. 

This also provides users confidence on what information is being collected on them, and sold. Namely, that if they firewalled the local instance, they can safely assume that none is being logged and sent away. As it stands now, doing so creates a minefield, of menu items that mislead people into thinking that Musescore can do something, which it cannot. 

Case in point, `import a PDF` is something that can be accomplished with local tools like OpenCV, for vector PDFs with even more efficient tools, and only optionally can be accomplished with a neural network with hardware requirements that are intractable for local use. While I appreciate that Musescore believes that the more computationally expensive method may yield a better objective result, the sheer fact that I would have to attach it to an online account makes it unacceptable. 

### Prior art

The Ollama project allows one to self-host large language models. 

The TabbyML project allows exposing a Copilot-like endpoint to code-oriented LLMs to be run locally. 

Almost all other musical applications that are licenced under an OSI approved licence, have opt-in, rather than opt-out online features that are not interspersed with offline features. 

Blender does not require an online account for most of its operation.

### Additional context

I am conscious of the fact that MuseScore is using the online account as its revenue stream. 

I would be happy to pay for a self-hosted version of MuseScore a fair market price. Finale, which is not an Open Source application, inspires more confidence, because the import of an image can be accomplished locally. If MuseScore were to have a way to self-host, I would be happy to pay the price of Finale 27 directly. 

### Checklist

- [X] This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- [X] I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

**Comment by bkunda** [comment]
Being an open source project, this is something that anyone who wants to is theoretically able to do.
So it's not really a feature request. 
Thanks for your understanding.

**Comment by appetrosyan** [comment]
> So it's not really a feature request.

You are correct. It was a provocation and you fell for it. 

This reply is going to be quoted as a specific example of a case of Open Source washing, where a commercial application with no respect for user freedom or privacy is actiuvely masquarading as such. 

> Being an open source project, this is something that anyone who wants to is theoretically able to do.

This is not an open source software project, nor is it free. It is a front-end to an opaque proprietary system that happens to be based on GPLv3 source code. While it is rather more convenient for you to say that it is an open source project, it is demeaning to thousands of people that have contributed code to projects that were actually done in accordance with the principles. 

> Thanks for your understanding.

Thank you for getting back to me so quickly. I almost thought I would have to push the article back, until I had solid evidence for reckless disregard of user privacy, you gave it to me on a very short notice, so I can get started right away. 

---- 

Sorry for acting in bad faith.... But to be quite honest, your project kinda has it coming.

**Comment by Tantacrul** [comment]
@appetrosyan 

You're wrong in a number of your assertions. Like, totally incorrect. Feel free to message me on info.tantacrul@gmail.com and I'll lay things out for you.

- **To your accusation that we don't respect privacy**. Actually, I personally put in a ton of effort into making sure that we improved upon the standards of previous versions of MuseScore we inherited from the founders. We don't track anything in the app.
